packages/server/src/jsRunner/vm/isolated-vm.ts

```diff
@@ -98,6 +98,20 @@ export class IsolatedVM implements VM {
     return this
   }
 
+  withSnippets(snippets?: Snippet[]) {
+    const snippetsSource = loadBundle(BundleType.SNIPPETS)
+    const script = this.isolate.compileScriptSync(`
+      const snippetDefinitions = ${JSON.stringify(snippets || [])};
+      ${snippetsSource};
+      snippets = snippets.default;
+    `)
+    script.runSync(this.vm, { timeout: this.invocationTimeout, release: false })
+    new Promise(() => {
+      script.release()
+    })
+    return this
+  }
```

Comment:
Snippet definitions are stringified and embedded into the isolate. We later read these definitions using the new snippets IVM bundle, to execute at runtime. By compiling the definitions into the isolate rather than adding them via context, like the client does, we can reduce the overall size of context and benefit from the fact we cache the isolate.